src/php/parser/Parser.php

```diff
@@ -0,0 +1,530 @@
+            }
+            case ExpressionType::NT:
+            {
+                $expression = Expression::asNonTerminalExpression($expression);
+                $name = $expression->getName();
+                $automaton = $this->automata[$name];
+                $cons = Continuations::cons(new NonTerminalContinuation($automaton->getMode(), $name, $asts, $error->getCurrentNonTerminal(), $position), $continuations);
+
+                return $this->evalNext($automaton->getExpression(), $position, new ASTs(), new RawError($error->getPosition(), $error->getNonTerminals(), $error->getFailedChars(), $name), $cons);
+            }
+            default:
+            {
+                $type = $expression->getType();
+                throw new RuntimeException("Unsupported expression type $type!");
+            }
+        }
+    }
+
+    private function moveApply(MatchResult $value, Continuation $evaluated, Continuations $rest): Action
+    {
+        switch ($value->getType()) {
+            case MatchResultType::ACCEPTED:
+            {
+                return $this->moveApplyOnAccept(Accepted::asAccepted($value), $evaluated, $rest);
+            }
+            case MatchResultType::REJECTED:
+            {
+                return $this->moveApplyOnReject(Rejected::asRejected($value), $evaluated, $rest);
+            }
+            default:
+            {
+                throw new RuntimeException("Unsupported type of result $value.");
+            }
+        }
+    }
+
+    private function moveApplyOnAccept(Accepted $accepted, Continuation $evaluated, Continuations $rest): Action
+    {
+        switch ($evaluated->getType()) {
+            case ContinuationType::SEQ:
+            {
+                $expressions = SeqContinuation::asSeqContinuation($evaluated)->getExpressions();
+                if ($expressions->isEmpty()) {
+                    return $this->applyNext($rest, $accepted);
+                } else {
+                    $cons = Continuations::cons(new SeqContinuation($expressions->tail()), $rest);
+                    return $this->evalNext($expressions->head(), $accepted->getPosition(), $accepted->getAsts(), $accepted->getError(), $cons);
+                }
+                break;
+            }
+            case ContinuationType::STAR:
+            case ContinuationType::PLUS:
+            {
+                $cons = Continuations::cons(new StarContinuation($evaluated->getExpression(), $accepted->getPosition(), $accepted->getAsts()), $rest);
+                return $this->evalNext($evaluated->getExpression(), $accepted->getPosition(), $accepted->getAsts(), $accepted->getError(), $cons);
+            }
+            case ContinuationType::ALT:
+            case ContinuationType::OPT:
+            {
+                return $this->applyNext($rest, $accepted);
+            }
+            case ContinuationType:: AND:
```

Comment:
Redundant space between `::` and `AND`

Reply:
resolved in d9e587ee5bc4b9568109af96eb9eb692c762e509